certora/harness/MarketplaceHarness.sol

```diff
@@ -6,23 +6,145 @@ import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
 import {IGroth16Verifier} from "../../contracts/Groth16.sol";
 import {MarketplaceConfig} from "../../contracts/Configuration.sol";
 import {Marketplace} from "../../contracts/Marketplace.sol";
-import {RequestId, SlotId} from "../../contracts/Requests.sol";
-import {Requests} from "../../contracts/Requests.sol";
+import {Requests, Request, RequestState, SlotState, RequestId, SlotId} from "../../contracts/Requests.sol";
 
 contract MarketplaceHarness is Marketplace {
-    constructor(MarketplaceConfig memory config, IERC20 token, IGroth16Verifier verifier)
-        Marketplace(config, token, verifier)
-    {}
+  using Requests for Request;
 
-    function publicPeriodEnd(Period period) public view returns (uint256) {
-        return _periodEnd(period);
+  constructor(
+    MarketplaceConfig memory config,
+    IERC20 token,
+    IGroth16Verifier verifier
+  ) Marketplace(config, token, verifier) {}
+
+  function publicPeriodEnd(Period period) public view returns (uint256) {
+    return _periodEnd(period);
+  }
+
+  function slots(SlotId slotId) public view returns (Slot memory) {
```

Comment:
Should this be called `slot` instead of `slots`?

Reply:
Well not really part of this PR, but yeah 😅